File: libsignal/service/src/test/java/org/whispersystems/signalservice/internal/push/ContentRange_parse_withInvalidStrings_Test.java
package org.whispersystems.signalservice.internal.push;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.junit.Assert.assertFalse;

@RunWith(Parameterized.class)
public final class ContentRange_parse_withInvalidStrings_Test {

  @Parameterized.Parameter(0)
  public String input;

  @Parameterized.Parameters(name = "Content-Range: \"{0}\"")
  public static Collection<Object[]> data() {
    return Arrays.asList(new Object[][]{
      { null },
      { "" },
      { "23-45/67" },
      { "ersions 23-45/67" },
      { "versions 23-45" },
      { "versions a-b/c" }
    });
  }

  @Test
  public void parse_should_be_absent() {
    assertFalse(ContentRange.parse(input).isPresent());
  }
}

File: libsignal-service/src/test/java/org/whispersystems/signalservice/internal/push/ContentRange_parse_withInvalidStrings_Test.kt
package org.whispersystems.signalservice.internal.push

import assertk.assertThat
import assertk.assertions.isFalse
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class ContentRange_parse_withInvalidStrings_Test(private val input: String?) {
  @Test
  fun parse_should_be_absent() {
    assertThat(ContentRange.parse(input).isPresent).isFalse()
  }

  companion object {
    @JvmStatic
    @Parameterized.Parameters(name = "Content-Range: \"{0}\"")
    fun data(): List<Array<String?>> {
      return listOf(
        arrayOf(null),
        arrayOf(""),
        arrayOf("23-45/67"),
        arrayOf("ersions 23-45/67"),
        arrayOf("versions 23-45"),
        arrayOf("versions a-b/c")
      )
    }
  }
}
